pkg/validating-webhooks/hive/v1/clusterdeployment_validating_admission_hook.go

```diff
@@ -280,25 +280,10 @@ func (a *ClusterDeploymentValidatingAdmissionHook) validateCreate(admissionSpec
 		}
 	}
 
-	if !cd.Spec.Installed {
-		if cd.Spec.Provisioning != nil {
-			if cd.Spec.Provisioning.InstallConfigSecretRef == nil || cd.Spec.Provisioning.InstallConfigSecretRef.Name == "" {
-				// InstallConfigSecretRef is not required for agent install strategy
-				if cd.Spec.Provisioning.InstallStrategy == nil || cd.Spec.Provisioning.InstallStrategy.Agent == nil {
-					allErrs = append(allErrs, field.Required(specPath.Child("provisioning", "installConfigSecretRef", "name"), "must specify an InstallConfig"))
-				}
-			}
-
-			// validate the agent install strategy:
-			if cd.Spec.Provisioning.InstallStrategy != nil &&
-				cd.Spec.Provisioning.InstallStrategy.Agent != nil {
-				allErrs = append(allErrs, validateAgentInstallStrategy(specPath, cd)...)
-			} else if cd.Spec.Platform.AgentBareMetal != nil {
-				// agent bare metal platform can only be used with agent install strategy:
-				allErrs = append(allErrs,
-					field.Forbidden(specPath.Child("platform", "agentBareMetal"),
-						"agent bare metal platform can only be used with agent install strategy"))
-			}
+	if !cd.Spec.Installed && cd.Spec.Provisioning != nil {
+		// InstallConfigSecretRef is not required for anyone using the new ClusterInstall interface:
+		if (cd.Spec.Provisioning.InstallConfigSecretRef == nil || cd.Spec.Provisioning.InstallConfigSecretRef.Name == "") && cd.Spec.ClusterInstallRef == nil {
```

Comment:
i don't think we should need to support the case where people are setting the provision without install config but also setting cluster install ref.

with set the provision section correctly or don't add it.

Reply:
so drop the `&& cd.Spec.ClusterInstallRef == nil` imo